datastore/datastore.nim

```diff
@@ -12,6 +13,9 @@ push: {.upraises: [].}
 
 type
   BatchEntry* = tuple[key: Key, data: seq[byte]]
+  Function*[T, U] = proc(value: T): U {.upraises: [CatchableError], gcsafe, closure.}
```

Comment:
We shouldn't use `upraises` anymore, just use the bare `raises: []`, upraises was a compatibility workaround for versions prior to 1.6.14, which didn't support `raises: []` and always needed an exception such as `raises: [Defect]`.

Maybe can be done in a followup cleanup, if there are more upraises used elsewhere in the stores (can't remember now).

Reply:
Thanks for the info. I simply changed it to `raises: [CatchableError]` and the code is compiling and tests are passing, so I think it's working as intended.